src/main/scala/higherkindness/rules_scala/workers/zinc/common/LoggedReporter.scala

```diff
@@ -0,0 +1,135 @@
+package higherkindness.rules_scala
+package workers.zinc.common
+
+// The following code is based on scala.tools.nsc.reporters.{AbstractReporter, ConsoleReporter, Reporter}
+// Copyright 2002-2009 LAMP/EPFL
+// see licenses/LICENSE_Scala
+// Original author: Martin Odersky
```

Comment:
How much of this code is from scala.tools.nsc.reporters? Perhaps we can move this to the thirdparty directory and add an appropriate notice.

Reply:
https://github.com/sbt/zinc/blob/develop/internal/zinc-compile-core/src/main/scala/sbt/internal/inc/LoggedReporter.scala
This is where we took the file from. We made some changes so they aren't exactly the same.

Reply:
Perhaps we could remove this comment? While, 98 101 105 106 107 are the only lines that I changed.

Reply:
As it stands, merely copying in this code is a clear violation of the original license(s). We'll need to follow the guidelines for the original source of this code from the Scala compiler _and_ and any additional constraints/licensing for the derivative work in Zinc. 

We have two options, as far as I see it:

- If you want to add this code to this repo we will need to provide a copy of Scalac and/or Zinc's LICENSE, with this derived code clearly attributed to those sources and with references to the licenses. We must follow the rules set out for the code we're copying in.

- Alternatively we can write our own reporter logic and avoid any OSS license SNAFUs.

Reply:
I'm all for writing our own. It's not a ton of code and doesn't seem too complex. What are people's thoughts about that? 

I'd rather avoid any license SNAFUs if we can.

Reply:
I'd prefer that we write our own, but I won't block this PR on preference alone.

Reply:
I also prefer to write our own. We could have more flexibility later on. I will work on it.